packages/nimble-components/src/wafer-map/specs/features/grid-dimension.md

```diff
@@ -0,0 +1,26 @@
+# Nimble Wafer Map Grid Dimensions
+
+## Problem Statement
+
+The Wafer Map displays dies on a canvas inside a circular border based on an internal grid which is calculated based on the data input.
+This approach is limiting the performance of the wafer and the grid dimensions can be erroneous if the displayed data is just a subset of a larger data set.
+To mitigate these problems, we will reveal three new API inputs which will allow the users to set the grid dimensions themselves, thus saving processing time and correcting inconsistencies generated by the lack of data context.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Wafer Map Spec](../README.md)
+-   [Interactive Wafer Map Improvements](https://dev.azure.com/ni/DevCentral/_wiki/wikis/Stratus/66917/Interactive-Wafer-Map-Improvements)
+
+## Implementation / Design
+
+The new attributes and properties will be described in the component definition. They will be created with change listeners which will trigger the appropriate updates in the wafer.
+
+The inputs will be validated before use. If they are undefined or the data is missing the previous behavior of calculating the values based on the input data will be implemented.
+
+## Alternative Implementations / Designs
+
+The API can be changed to use four numeric inputs for each of the values, or a new Type can be created to structure all the dimensions together.
```

Comment:
Could you also describe why these alternatives were rejected in favor of the proposed approach?

Reply:
They are not rejected, sorry if that is the way it appears. I chose this approach because the PointCoordinates is already a public type, merging all 4 of them in 1 property will need a new public type, and having them all as attributes will increase the chance of missing information, but they are all valid in my opinion